Flaky tests: DaemonTests::test_5_daemon_sigint variance in return code

See:
```
5/9 Test #5: python_test_osqueryd .............***Failed   33.75 sec
.I0721 12:22:59.882045 26907 options.cpp:61] Verbose logging enabled by config option
I0721 12:22:59.882532 26907 daemon.cpp:38] Not starting the distributed query service: Distributed query service not enabled.
...FI0721 12:23:29.781241 26946 options.cpp:61] Verbose logging enabled by config option
I0721 12:23:29.781740 26946 daemon.cpp:38] Not starting the distributed query service: Distributed query service not enabled.
.
======================================================================
FAIL: test_5_daemon_sigint (__main__.DaemonTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/osquery/jenkins/workspace/osqueryPullRequestBuild/TargetSystem/centos7/tools/tests/test_base.py", line 455, in wrapper
    raise exceptions[0][0]
AssertionError: -2 != 130

----------------------------------------------------------------------
Ran 6 tests in 33.695s

FAILED (failures=1)
Test (attempt 1) DaemonTests::test_5_daemon_sigint failed: -2 != 130 (test_base.py:437)
Test (attempt 2) DaemonTests::test_5_daemon_sigint failed: -2 != 130 (test_base.py:437)
Test (attempt 3) DaemonTests::test_5_daemon_sigint failed: -2 != 130 (test_base.py:437)
```

For an example see: https://jenkins.osquery.io/job/osqueryPullRequestBuild/2912/TargetSystem=centos7/console

--- a/tools/tests/test_base.py
+++ b/tools/tests/test_base.py
@@ -437,9 +437,10 @@ def flaky(gen):
             worked = gen(this)
             return True
         except Exception as e:
-            exc_type, exc_obj, exc_tb = sys.exc_info()
-            fname = os.path.split(exc_tb.tb_frame.f_code.co_filename)[1]
-            exceptions.append((e, fname, exc_tb.tb_lineno))
+            import traceback
+            exc_type, exc_obj, tb = sys.exc_info()
+            exceptions.append((e))
+            print (traceback.format_tb(tb)[1])
         return False
     def wrapper(this):
         for i in range(3):
@@ -447,12 +448,12 @@ def flaky(gen):
                 return True
         i = 1
         for exc in exceptions:
-            print("Test (attempt %d) %s::%s failed: %s (%s:%d)" % (
+            print("Test (attempt %d) %s::%s failed: %s" % (
                 i,
                 this.__class__.__name__,
-                gen.__name__,  str(exc[0]), exc[1], exc[2]))
+                gen.__name__,  str(exc[0])))
             i += 1
-        raise exceptions[0][0]
+        raise Exception(exceptions[0][0])
     return wrapper
 
 

--- a/tools/tests/test_osqueryd.py
+++ b/tools/tests/test_osqueryd.py
@@ -90,13 +90,16 @@ class DaemonTests(test_base.ProcessGenerator, unittest.TestCase):
         # An interrupt signal will cause the daemon to stop.
         daemon = self._run_daemon({
             "disable_watchdog": True,
+            "ephemeral": True,
+            "disable_database": True,
+            "disable_logging": True,
         })
         self.assertTrue(daemon.isAlive())
 
         # Send a SIGINT
         os.kill(daemon.pid, signal.SIGINT)
         self.assertTrue(daemon.isDead(daemon.pid, 10))
-        self.assertEqual(daemon.retcode, 128 + signal.SIGINT)
+        self.assertTrue(daemon.retcode in [128 + signal.SIGINT, -2])
 
     @test_base.flaky
     def test_6_logger_mode(self):

